apps/arweave/src/ar_serialize.erl

```diff
@@ -1820,3 +1825,214 @@ decode_if_set(JSON, JSONProperty, Decoder, Default) ->
 		EncodedValue ->
 			Decoder(EncodedValue)
 	end.
+
+jobs_to_json_struct(Jobs) ->
+	#jobs{ vdf = JobList, diff = Diff, seed = Seed, next_seed = NextSeed,
+			interval_number = IntervalNumber, next_vdf_difficulty = NextVDFDiff } = Jobs,
+
+	{[{vdf, [job_to_json_struct(Job) || Job <- JobList]},
+		{diff, integer_to_binary(Diff)},
+		{seed, ar_util:encode(Seed)},
+		{next_seed, ar_util:encode(NextSeed)},
+		{interval_number, integer_to_binary(IntervalNumber)},
+		{next_vdf_difficulty, integer_to_binary(NextVDFDiff)}
+	]}.
+
+job_to_json_struct(Job) ->
+	#job{ output = Output, global_step_number = StepNumber,
+			partition_upper_bound = PartitionUpperBound } = Job,
+	{[{output, ar_util:encode(Output)},
+			{global_step_number, integer_to_binary(StepNumber)},
+			{partition_upper_bound, integer_to_binary(PartitionUpperBound)}]}.
+
+json_struct_to_jobs(Struct) ->
+	{Keys} = Struct,
+	Diff = binary_to_integer(proplists:get_value(<<"diff">>, Keys, <<"0">>)),
+	Seed = ar_util:decode(proplists:get_value(<<"seed">>, Keys, <<>>)),
+	NextSeed = ar_util:decode(proplists:get_value(<<"next_seed">>, Keys, <<>>)),
+	NextVDFDiff = binary_to_integer(proplists:get_value(<<"next_vdf_difficulty">>, Keys,
+			<<"0">>)),
+	IntervalNumber = binary_to_integer(proplists:get_value(<<"interval_number">>, Keys,
+			<<"0">>)),
+	VDF = [json_struct_to_job(Job) || Job <- proplists:get_value(<<"vdf">>, Keys, [])],
+	#jobs{ vdf = VDF, seed = Seed, next_seed = NextSeed, interval_number = IntervalNumber,
+			next_vdf_difficulty = NextVDFDiff, diff = Diff }.
+
+json_struct_to_job(Struct) ->
+	{Keys} = Struct,
+	Output = ar_util:decode(proplists:get_value(<<"output">>, Keys, <<>>)),
+	StepNumber = binary_to_integer(proplists:get_value(<<"global_step_number">>, Keys,
+			<<"0">>)),
+	PartitionUpperBound = binary_to_integer(proplists:get_value(<<"partition_upper_bound">>,
+			Keys, <<"0">>)),
+	#job{ output = Output, global_step_number = StepNumber,
+			partition_upper_bound = PartitionUpperBound }.
+
+partial_solution_to_json_struct(
```

Comment:
Since a "solution" is also a "partial solution" (I think?) can we remove this distinction an only have a single solution serializer/deserializer? (e.g. remove `solution_to_json_struct` or remove `partial_solution_to_json_struct`)

Reply:
Done